xbmc/addons/gui/skin/SkinTimerManager.cpp

```diff
@@ -23,39 +23,39 @@ using namespace std::chrono_literals;
 
 void CSkinTimerManager::LoadTimers(const std::string& path)
 {
-  CXBMCTinyXML doc;
+  CXBMCTinyXML2 doc;
   if (!doc.LoadFile(path))
   {
-    CLog::LogF(LOGWARNING, "Could not load timers file {}: {} (row: {}, col: {})", path,
-               doc.ErrorDesc(), doc.ErrorRow(), doc.ErrorCol());
+    CLog::LogF(LOGWARNING, "Could not load timers file {}: {} (Line: {})", path, doc.ErrorStr(),
+               doc.ErrorLineNum());
     return;
   }
 
-  TiXmlElement* root = doc.RootElement();
+  auto* root = doc.RootElement();
   if (!root || !StringUtils::EqualsNoCase(root->Value(), "timers"))
   {
     CLog::LogF(LOGERROR, "Error loading timers file {}: Root element <timers> required.", path);
     return;
   }
 
-  const TiXmlElement* timerNode = root->FirstChildElement("timer");
+  const auto* timerNode = root->FirstChildElement("timer");
   while (timerNode)
   {
     LoadTimerInternal(timerNode);
     timerNode = timerNode->NextSiblingElement("timer");
   }
 }
 
-void CSkinTimerManager::LoadTimerInternal(const TiXmlElement* node)
+void CSkinTimerManager::LoadTimerInternal(const tinyxml2::XMLNode* node)
 {
-  if ((!node->FirstChild("name") || !node->FirstChild("name")->FirstChild() ||
-       node->FirstChild("name")->FirstChild()->ValueStr().empty()))
+  if (!node->FirstChildElement("name") || !node->FirstChildElement("name")->FirstChild() ||
+       (strcmp(node->FirstChildElement("name")->FirstChild()->Value(), "\0") != 0))
   {
     CLog::LogF(LOGERROR, "Missing required field name for valid skin. Ignoring timer.");
```

Comment:
While at it maybe you can fix the log line?
```suggestion
    CLog::LogF(LOGERROR, "Missing required field name for valid skin timer. Ignoring timer.");
```

Reply:
nevermind, did it in https://github.com/xbmc/xbmc/pull/23608